File: Praktikum2/Java/Memory.java
class Memory extends Hardware {
    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }

    public String getFrequency() {
        return this.frequency;
    }

    public void setMemorySize(int memorySize) {
        this.memorySize = memorySize;
    }

    public int getMemorySize() {
        return this.memorySize;
    }

    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    public String getSupportsCuda() {
        return this.supportsCuda;
    }

    public void printMemory() {
        System.out.println("Frequency : " + this.frequency);
        System.out.println("Memory Size : " + this.memorySize);
        System.out.println("Support Cuda : " + this.supportCuda);
    }
}

File: Praktikum2/C++/Memory.cpp
#include <bits/stdc++.h>
#include "Hardware.cpp"
using namespace std;

class Memory : public Hardware
{
private:
    string frequency;
    int memorySize;
    string supportsCuda;

public:
    Memory() {}

    void setFrequency(string frequency)
    {
        this->frequency = frequency;
    }

    string getFrequency()
    {
        return this->frequency;
    }

    void setMemorySize(int memorySize)
    {
        this->memorySize = memorySize;
    }

    int getMemorySize()
    {
        return this->memorySize;
    }

    void setSupportsCuda(string supportsCuda)
    {
        this->supportsCuda = supportsCuda;
    }

    string getSupportsCuda()
    {
        return this->supportsCuda;
    }

    void printMemory()
    {
        cout << "Frequency : " << this->getFrequency() << endl;
        cout << "Memory Size : " << this->getMemorySize() << endl;
        cout << "Support Cuda : " << this->getSupportsCuda() << endl;
    }

    ~Memory() {}
};